src/hooks/analytics/useTrackWalletConnectSessionEstablished.ts

```diff
@@ -0,0 +1,30 @@
+import { usePostHog } from 'posthog-react-native';
+import React from 'react';
+import { SessionTypes } from '@walletconnect/types';
+
+const EVENT_WALLET_CONNECT_SESSION_ESTABLISHED = 'WalletConnect Session Established';
+
+/**
+ * Hook that provides a function to track when a user approve a
+ * WalletConnect session request.
+ */
+const useTrackWalletConnectSessionEstablished = () => {
+  const postHog = usePostHog();
+
+  return React.useCallback(
+    (session: SessionTypes.Struct) => {
+      if (!postHog) {
+        return;
+      }
+
+      postHog.capture(EVENT_WALLET_CONNECT_SESSION_ESTABLISHED, {
+        CreationTime: new Date().toISOString(),
+        ApplicationName: session.peer.metadata.name,
+        Namespaces: session.requiredNamespaces,
```

Comment:
@bragaz I decided to not track the description of the WalletConnect app since can be useless, let me know if you want me to track it.